crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -161,12 +164,21 @@ public Response insert(InsertionRequest req) {
                 ctx.getHookManager().callQueuedHooks();
             }
         } catch (Error e) {
+            if(metricCtx!=null) {
+                metricCtx.markRequestException(e);
+            }
             response.getErrors().add(e);
             response.setStatus(OperationStatus.ERROR);
         } catch (Exception e) {
+            if(metricCtx!=null) {
+                metricCtx.markRequestException(e);
+            }
             response.getErrors().add(Error.get(CrudConstants.ERR_CRUD, e));
             response.setStatus(OperationStatus.ERROR);
         } finally {
+            if(metricCtx!=null) {
+                metricCtx.endRequestMonitoring();
```

Comment:
Some exceptions might not get metered if they don't bubble up to the catch blocks here. So maybe what we want to do is, instead of adding to each catch block, rely on the errors being added to the response as they should be. Then, in the finally block, go through each of the errors in the Response and mark them with the request ctx. This ensures we are tracking response errors thoroughly.